### PR TITLE
Key strategy added (encoding/decoding)

### PR DESCRIPTION
**Pull request by romansladecekinloop**
Possibility to change KeyEncodingStrategy / KeyDecodingStrategy (e.g. sneakCase)

### CODE REVIEW

File: Example/Tests/EnvironmentTests.swift [hunk 1]
```diff
@@ -64,6 +68,15 @@ struct GithubError: Decodable, CustomDateDecodable {
     }
 
     public static var dateDecodingStrategy = JSONDecoder.DateDecodingStrategy.iso8601
+    static var keyDecodingStrategy: JSONDecoder.KeyDecodingStrategy = .useDefaultKeys
+}
+
+struct GithubErrorSneakCase: Decodable, CustomDateDecodable {
```
Comment (radimhalfar) [inline Example/Tests/EnvironmentTests.swift]: It's snakeCase

File: Example/Tests/EnvironmentTests.swift [hunk 2]
```diff
@@ -282,6 +295,28 @@ class EnvironmentTests: QuickSpec {
                 expect(githubAdapter.errorsHandled).to(equal(1))
             }
 
+            it("error parsing (sneak case)") {
+                var errorMessage: String? = nil
+                let endpoint = GithubEnvironment.blah
```
Comment (radimhalfar) [inline Example/Tests/EnvironmentTests.swift]: Don't we have something more representative than `blah` ?

**Review by radimhalfar — CHANGES_REQUESTED** [review]

File: Example/Tests/EnvironmentTests.swift (hunk 1 above)
Comment (jakubpetrik) [reply Example/Tests/EnvironmentTests.swift]: that is one sneaky typo. 😉

File: Example/Tests/EnvironmentTests.swift [hunk 3]
```diff
@@ -282,6 +290,28 @@ class EnvironmentTests: QuickSpec {
                 expect(githubAdapter.errorsHandled).to(equal(1))
             }
 
+            it("error parsing (sneak case)") {
```
Comment (radimhalfar) [inline Example/Tests/EnvironmentTests.swift]: typo

**Review by radimhalfar — APPROVED** [review]

**Pull request merged**